src/videoframeprocessor/VideoFrameProcessor.ts

```diff
@@ -0,0 +1,26 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import VideoFrameBuffer from './VideoFrameBuffer';
+
+/**
+ * [[VideoFrameProcessor]] is defined as a unit of video processing.
+ * It transforms an array of {@link VideoFrameBuffer} and produces an array of {@link VideoFrameBuffer}.
+ * It can be chained together to be used with {@link VideoFrameProcessorPipeline}.
+ */
+export default interface VideoFrameProcessor {
+  /**
+   * Returns the name of the processor
+   */
+  name(): string;
+
```

Comment:
Is this used?

Reply:
not in demo. I kind of like processor has identifier but I could remove it if you think it's too extra.

Reply:
I think a processor could implement `toString` if it wants to, right? If you're logging or otherwise inspecting this thing, you might want to print `<Crop 80x80>`, no?

This is the danger of defining APIs without having some concrete uses in mind — `name` might be too rigid a concept.

Reply:
I see. Make sense. Removed